code/form/FoxyStripeStoreDomainField.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+/**
+ * based on SiteTreeURLSegmentField
+ */
+class FoxyStripeStoreDomainField extends TextField
+{
+    /**
+     * @var string
+     */
+    protected $helpText;
+    protected $urlSuffix = '.foxycart.com';
+
+    private static $allowed_actions = array(
+        'suggest'
+    );
+
+    public function Value()
+    {
+        return rawurldecode($this->value);
+    }
+
+    public function Field($properties = array())
+    {
+        Requirements::javascript(MODULE_FOXYSTRIPE_DIR . '/javascript/FoxyStripeStoreDomainField.js');
+        Requirements::css(MODULE_FOXYSTRIPE_DIR . "/css/foxystripescreen.css");
+        return parent::Field($properties);
+    }
+
+    public function suggest($request)
+    {
+        if (!$request->getVar('value')) {
+            return $this->httpError(405,
+                _t('SiteTreeURLSegmentField.EMPTY', 'Please enter a URL Segment or click cancel')
+            );
+        }
+
+        $value = $request->getVar('value');
+
+        $checkName = function () use (&$value, &$checkName) {
```

Comment:
This will need refactoring as the api call is in the closure. We should break this out so it can be used by other pars of foxystripe as we continue to integrate the hAPI.